### PR TITLE
add: BearerAuthentication middleware

### PR DESCRIPTION
**Pull request by devcxl**
Adding the `BearerAuthentication` middleware enhances the security of the interface.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,31 +10,41 @@ import { getImageHandler, imageGenerationHandler } from './routes/image';
...
+	if (authorizationHeader) {
+		const parts = authorizationHeader.split(' ');
+		if (parts.length === 2 && parts[0] === 'Bearer') {
+			return parts[1];
+		}
+	}
+	return null;
+}
+// MIDDLEWARE: withAuthenticatedUser - embeds user in Request or returns a 401
+const bearerAuthentication = (request, env) => {
```
Comment (chand1012) [inline index.js:25]: I think this is a great change, however I think we should have the option to leave the `ACCESS_TOKEN` blank and to then ignore this middleware if and only if the token is omitted from the config.

File: index.js [hunk 2]
```diff
@@ -10,31 +10,41 @@ import { getImageHandler, imageGenerationHandler } from './routes/image';
 const { preflight, corsify } = createCors();
 
 // Create a new router
-const router = Router();
+const router = Router({ base: '/v1' });
```
Comment (chand1012) [inline index.js:13]: The base should be configurable via environment variable and the default should be blank, as in no added base URL.

**Review by chand1012 — CHANGES_REQUESTED** [review]
I think overall these are good changes, however there are two small tweaks that I would make. Also thanks for the refactor of the endpoints, looks much cleaner now!

**Comment by devcxl** [comment]
I think it's necessary to enforce the addition of ACCESS_TOKEN to ensure the security of the interface. As for the suggestion to control the setting of the base using environment variables, it's indeed a good idea.

**Comment by devcxl** [comment]
However, how to pass the environment variables to the base is a problem.

**Comment by chand1012** [comment]
That's fine then, could you modify the README to add instructions about changing the ACCESS_TOKEN so that people don't get confused when they get rejected by their instance? Thanks.

File: wrangler.toml [hunk 3]
```diff
@@ -12,6 +15,10 @@ binding = "IMAGE_BUCKET"
 bucket_name = "openai-cf-images"
 preview_bucket_name = "ai-images"
 
+[vars]
+ACCESS_TOKEN = "sk-1234567890ASDFGHJKLQWERTYUIOP" # You can set any string value."
```
Comment (chand1012) [inline wrangler.toml:19]: Need instructions about this in the README.

**Review by chand1012 — CHANGES_REQUESTED** [review]
Once instructions are added I'll merge.

**Comment by devcxl** [comment]
Okay, I'll update the README file later.

**Comment by devcxl** [comment]
Done

**Review by chand1012 — APPROVED** [review]

**Comment by chand1012** [comment]
Thank you for your help!

**Pull request merged**